### DISCUSSION TITLE
[J4] Multiple Module Positions for Modules

### DISCUSSION THREAD

**Discussion by dautrich** [joomla/joomla-cms#31767] [2. Cold Features]
**Summary:**
Currently, a module can be assigned to a maximum of one module positions, regardless of how many templates are installed. I propose to allow one module position **_per_** template installed.

**Motivation:**
There are a number of use cases where the current solution is way too inflexible and leads to a lot of unnecessary work for webmasters, especially if the site has numerous modules.

**Use Cases:**
1. When changing a website to a new template, a user should be able to assign his modules to the module positions of the new template without breaking the existing structure. Comparing the old and the new template for the specific site would become way easier.
2. If a user has discovered an issue, which might be related to the template used, supporters in Joomla forums or user groups will propose: “Test the issue with Protostar!”. If the administrator wants to follow this suggestion, he will have to break the structure of his site (see use case 1).
3. A webmaster wants to use two different templates within one site, structured by menu assignment, but needs modules in pages from both templates. Today he will have to create duplicate content or use some other trick.

**Implication:** 
Currently, there is a 1:1 association between module and module position. This structure should be changed to a 1:n association.

**Comment by brianteeman** [comment]
All you have to d is to change the name of the modules in the template to match

**Reply by dautrich** [reply]
> All you have to d is to change the name of the modules in the template to match

I don't think this solves the problem. Templates have different numbers of module positions, so an exchange of module names isn't sufficient. Also, most Joomlers won't be able to change the names of the module positions in a template.

**Reply by brianteeman** [reply]
It is a simple solution that works. The suggestions you propose will create a very cumbersome UI for an occasional use.

**Comment by dautrich** [comment]
> The suggestions you propose will create a very cumbersome UI for an occasional use.

With a little bit of luck there might be ideas to avoid cumbersome UIs. It could be solved in a similar way as in this template from Joomla51:

![Screenshot_2020-12-23 Templates Edit Style - Neudorf - Administration](https://user-images.githubusercontent.com/7933215/103028799-c20cf600-4558-11eb-81a3-c774fbe0ec5d.png)
